src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -174,6 +178,58 @@ async fn skip_cache_on_error() {
   assert_eq!(result.exit_code, 0);
 }
 
+/// With eager_fetch enabled, we should skip the remote cache if any of the process result's
+/// digests are invalid. This will force rerunning the process locally. Otherwise, we should use
+/// the cached result with its non-existent digests.
+#[tokio::test]
+async fn eager_fetch() {
+  let workunit_store = WorkunitStore::new(false);
+  workunit_store.init_thread_state(None);
+
+  async fn run_process(eager_fetch: bool) -> FallibleProcessResultWithPlatform {
+    let (local, store, _local_runner_dir, _stub_cas) = create_local_runner();
+    let (caching, _cache_dir, stub_action_cache) =
+      create_cached_runner(local, store.clone(), eager_fetch);
+
+    // Get the `action_digest` for the Process that we're going to run. This will allow us to
+    // insert a bogus value into the `stub_action_cache`.
+    let (process, _script_path, _script_dir) = create_script(1);
+    let (action, command, _exec_request) =
+      make_execute_request(&process, ProcessMetadata::default()).unwrap();
+    let (_command_digest, action_digest) = ensure_action_stored_locally(&store, &command, &action)
+      .await
+      .unwrap();
+
+    // Insert an ActionResult with missing digests and a return code of 0 (instead of 1).
+    let bogus_action_result = ActionResult {
+      exit_code: 0,
+      stdout_digest: Some(TestData::roland().digest().into()),
+      stderr_digest: Some(TestData::roland().digest().into()),
+      ..ActionResult::default()
+    };
+    stub_action_cache
+      .action_map
+      .lock()
+      .insert(action_digest.0, bogus_action_result);
+
+    // Run the process, possibly by pulling from the `ActionCache`.
+    caching
+      .run(process.clone().into(), Context::default())
+      .await
+      .unwrap()
+  }
+
+  let lazy_result = run_process(false).await;
+  assert_eq!(lazy_result.exit_code, 0);
+  assert_eq!(lazy_result.stdout_digest, TestData::roland().digest());
+  assert_eq!(lazy_result.stderr_digest, TestData::roland().digest());
+
+  let eager_result = run_process(true).await;
+  assert_eq!(eager_result.exit_code, 1);
+  assert_ne!(eager_result.stdout_digest, TestData::roland().digest());
+  assert_ne!(eager_result.stderr_digest, TestData::roland().digest());
```

Comment:
Maybe check the applicable counters in the workunit store are incremented depending on whether there was an ignored error or not?